src/main/java/de/lh/tool/service/entity/impl/DtoAssemblyServiceImpl.java

```diff
@@ -0,0 +1,115 @@
+
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import de.lh.tool.domain.dto.HelperTypeDto;
+import de.lh.tool.domain.dto.NeedDto;
+import de.lh.tool.domain.dto.NeedUserDto;
+import de.lh.tool.domain.dto.ProjectHelperTypeDto;
+import de.lh.tool.domain.dto.UserDto;
+import de.lh.tool.domain.dto.assembled.AssembledHelperTypeDto;
+import de.lh.tool.domain.dto.assembled.AssembledHelperTypeWrapperDto;
+import de.lh.tool.domain.dto.assembled.AssembledNeedDto;
+import de.lh.tool.domain.dto.assembled.AssembledNeedUserDto;
+import de.lh.tool.domain.dto.assembled.AssembledProjectHelperTypeDto;
+import de.lh.tool.domain.exception.DefaultException;
+import de.lh.tool.domain.exception.DefaultRuntimeException;
+import de.lh.tool.domain.exception.ExceptionEnum;
+import de.lh.tool.service.entity.interfaces.DtoAssemblyService;
+import de.lh.tool.service.entity.interfaces.HelperTypeService;
+import de.lh.tool.service.entity.interfaces.NeedService;
+import de.lh.tool.service.entity.interfaces.NeedUserService;
+import de.lh.tool.service.entity.interfaces.ProjectHelperTypeService;
+import de.lh.tool.service.entity.interfaces.UserService;
+
+@Service
+public class DtoAssemblyServiceImpl implements DtoAssemblyService {
+	@Autowired
+	private HelperTypeService helperTypeService;
+	@Autowired
+	private ProjectHelperTypeService projectHelperTypeService;
+	@Autowired
+	private NeedService needService;
+	@Autowired
+	private NeedUserService needUserService;
+	@Autowired
+	private UserService userService;
+
+	private ModelMapper modelMapper = new ModelMapper();
+
+	@Transactional
+	@Override
+	public Map<String, AssembledHelperTypeWrapperDto> findHelperTypesWithNeedsAndUsersBetweenDates(Long projectId,
+			LocalDate start, LocalDate end) throws DefaultException {
+		HashMap<String, AssembledHelperTypeWrapperDto> data = new HashMap<>();
+		LocalDate date = start;
+		while (date.isBefore(end) || date.isEqual(end)) {
+			final LocalDate finalDate = date;
+			data.put(date.toString(),
+					new AssembledHelperTypeWrapperDto(helperTypeService
+							.findDtosByProjectIdAndWeekday(projectId, finalDate.getDayOfWeek().getValue()).stream()
+							.map(ht -> assembleHelperType(projectId, ht, finalDate)).collect(Collectors.toList())));
+
+			date = date.plusDays(1);
+		}
+
+		return data;
+	}
+
+	private AssembledHelperTypeDto assembleHelperType(Long projectId, HelperTypeDto helperType, LocalDate date) {
+		AssembledHelperTypeDto assembled = modelMapper.map(helperType, AssembledHelperTypeDto.class);
+		try {
+			assembled.setShifts(projectHelperTypeService
+					.findDtosByProjectIdAndHelperTypeIdAndWeekday(projectId, helperType.getId(),
+							date.getDayOfWeek().getValue())
+					.stream().map(pht -> assembleProjectHelperType(pht, date)).collect(Collectors.toList()));
+		} catch (DefaultException e) {
+			throw new DefaultRuntimeException(e);
```

Comment:
Warum `DefaultException` nicht einfach hochbubbeln lassen?

`findHelperTypesWithNeedsAndUsersBetweenDates()` ist ja bereits mit `throws DefaultException` annotiert.

Reply:
Dann kann ich es nicht im Stream verwenden oder geht das auch mit einer normalen Exception?

Reply:
Du hast du natürlich recht. Die Signaturen von den Lambdas stimmen bei checked Exceptions dann nicht mehr überein.
Hier werden ein paar Vorschläge gemacht, wie man das kapseln könnte:
https://www.baeldung.com/java-lambda-exceptions

Muss imho aber auch nicht sein...

Reply:
Stimmt. So wie ich das verstehe läuft es in irgendeiner Form aber immer auf RTEs raus, wenn man die Exception nicht in irgendeiner Form behandeln möchte. Nachdem wir selber die Exceptions werfen und sie nur bis zum Rest-Controller durchreichen, sehe ich persönlich da jetzt auch keine Veranlassung, da extra einen Wrapper zu bauen.